launch/launch/actions/execute_process.py

```diff
@@ -577,6 +583,16 @@ def __expand_substitutions(self, context):
             self.__logger.info("process details: cmd=[{}], cwd='{}', custom_env?={}".format(
                 ', '.join(cmd), cwd, 'True' if env is not None else 'False'
             ))
+        try:
+            emulate_tty = yaml.safe_load(
+                context.launch_configurations['emulate_tty']
+            )
+            if type(emulate_tty) is not bool:
+                raise TypeError('emulate_tty is not boolean [{}]'.format(
+                    type(emulate_tty)
+                ))
+        except KeyError:
+            emulate_tty = self.__emulate_tty
```

Comment:
I'm usually OK with Python's style of simply letting things throw but in this case I think that checking for `'emulate_tty' in context.launch_configurations` instead would aid readability.

Reply:
Have a look at 1b84e42.